[knx] New modifier to set mainGA write-only (#16042)
openhab/openhab-addons: bundles/org.openhab.binding.knx/README.md

#### Group Address Notation

```text
<config>="[<dpt>:][<>]<mainGA>[[+[<]<listeningGA>][+[<]<listeningGA>..]]"
```

where parts in brackets `[]` denote optional information.

**Each configuration parameter has a `mainGA` where commands are written to and optionally several `listeningGA`s.**

`mainGA` also listens to incoming packets, unless prefixed with a `>` character.
This is recommended if you have dedicated status group addresses which are added as `listeningGA`s.

The optional `<` sign tells whether the group address of the datapoint accepts read requests on the KNX bus (it does, if the sign is there).

> Group addresses marked with `<` are read by openHAB during startup, but typically this is useful for only one group address per definition.

Reviewer: ```suggestion
The group addresses marked with `<` are read by openHAB during startup. 
Future versions might support reading from one GA only.
```

Or can you imagine a setup where reading several addresses for one state would be correct? The result would be non-deterministic, because the final state would depend on the device with the slowest response.

Author: No, I can't. The patch already introduces a warning on info level. The question is how to notify users who use the GUI.

Reviewer 2: I agree here.

>  The question is how to notify users who use the GUI.

That's a good question in many cases ... I can imagine to use the log WebSocket to register to the logger of the add-on of the Thing you currently edit and show some alert in the UI if a warning or error message is received over the WebSocket. But this is just a general idea for the future ... in the mean time, I would say, it's best to have a look at your logs when changing configuration. (I always watch my logs when changing configuration.)

With `*-control` channels, the state is not owned by any device on the KNX bus, therefore no read requests will be sent by the binding, i.e. `<` signs will be ignored for them.

> The element `dpt` is optional. If omitted, the corresponding default value will be used (see the channel descriptions above).

Reviewer: I wonder if we should change that. A lot of issues especially with DPT9 would not have occurred if the DPT was mandatory. I would prefer to remove the "optional" brackets around the DPT and change this sentence as suggested.

```suggestion
The element `dpt` is highly recommended and change to a mandatory element in future versions.
If omitted, the corresponding default value will be used (see the channel descriptions above). 
```

The code can be like it is now. In the next iteration (i.e. 4.2) we could print a warning for a missing DPT and remove support for "default" DPTs earliest in 4.3.

#### Group Address Notation

```text

> <config>="[<dpt>:][<>]<mainGA>[[+[<]<listeningGA>][+[<]<listeningGA>..]]"

Reviewer 3: Wouldn't the proper notation be
```
<config>="[<dpt>:][<|>]<mainGA>[[+[<]<listeningGA>][+[<]<listeningGA>..]]"
```

Reviewer 2: Yes, only one of both modifiers is „allowed“, but not sure if that is needed as only one makes sense.

In case we want to note that, I would prefer to have a short sentence saying that, because | is not-exclusive OR, but in fact it is < XOR >.

Reviewer 3: I know, it is not a formal syntax, but the ' |' is supposed to say it is one of the chars that goes there (if any). At the moment it looks as if it is optional to put `<>` in there (and nothing else).

Reviewer 2: An alternative would be to say `[<][>]` as is said that the parts in brackets are optional. For that we would definitely need to say it is either … or ….

WDYT is easier to read and more “intuitive” for the end user?

Reviewer 3: ' [<][>]` would clearly indicate that both are optional, but independent. That's why I suggested `[<|>]`, which is imho a quite common way to show the alternatives that can go into the optional placeholder.

Reviewer: Using `|` as "one or the other" is also the Java regex style, so I would vote for that.

Author: Maybe we should change the style to mark variables '''<dpt>''',  as the arrows have a different meaning.
dpt should do, and I think using | for or is intuitive for most users....

Reviewer 3: No, I would leave the variables as `<variable>` - imho this is very common and easy to understand.

```

where parts in brackets `[]` denote optional information.

**Each configuration parameter has a `mainGA` where commands are written to and optionally several `listeningGA`s.**

`mainGA` also listens to incoming packets, unless prefixed with a `>` character.

> This is recommended if you have dedicated status group addresses which are added as `listeningGA`s.

Reviewer 2: ```suggestion
This is recommended if you have a dedicated status group address which is added as `listeningGA`.
```

I can't imagine a case where a single channel config has more than a single status GA. Do you?
